File: backend/app/schemas/inspection.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class InspectionCreate(BaseModel):
    inspection_type: str

    product_name: Optional[str] = None
    brand_name: Optional[str] = None
    manufacturer_name: Optional[str] = None

    source_url: Optional[str] = None


class InspectionResponse(BaseModel):
    id: int
    inspection_number: str
    officer_id: int

    inspection_type: str

    product_name: Optional[str]
    brand_name: Optional[str]
    manufacturer_name: Optional[str]

    source_url: Optional[str]

    status: str
    overall_result: str

    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: frontend/legal-metrology-compliance/types/inspection.ts
export type InspectionCreate = {
  inspection_type: string;
  product_name?: string;
  brand_name?: string;
  manufacturer_name?: string;
  source_url?: string;
};

export type InspectionResponse = {
  id: number;
  inspection_number: string;
  officer_id: number;
  inspection_type: string;
  product_name: string | null;
  brand_name: string | null;
  manufacturer_name: string | null;
  source_url: string | null;
  status: string;
  overall_result: string;
  created_at: string;
  updated_at: string;
};
